### ISSUE TITLE
Limit the appearance of the Auto-Update Check Notification only if the Windows Desktop (explorer.exe) has focus

### ISSUE THREAD

**Issue by DrSooom** [nvaccess/nvda#9409]
### Steps to reproduce:
+ Wait for the auto-update check notification.
### Actual behaviour:
This notification seems to appear suddenly and no matter which application has currently the focus.
### Expected behaviour:
As this leads to unwanted interactions and notifications, because it interrupts the workflow suddenly, the auto-update check notification should only appear after the user switch to the Windows Desktop (explorer.exe). Of course it could be that this notification will still appear after opening/switching to the Windows Explorer itself too, because of technical limitations.
#### NVDA installed/portable:
Both
#### NVDA version:
The issue description depends on 2018.1, but it should be the same for 2019.1 too.
#### Windows version:
Win7 to Win10-1903
### See also:
Issue #8514, #9174 and #9408

**Comment by ehollig** [comment]
I am tempted to close this as duplicate of #2257

**Comment by Brian1Gaff** [comment]
Funnily enough I was just having a gentle moan about this on the user list. 
Exactly what is it that triggers it?
 I find I now use manual checking, but if such a system can be  made to 
function without harm to the system it seems like a worthwhile change. Same 
goes for add on updates. There is a further issue here as both can appear 
together and one stops the other as of course it reboots nvda each time.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by leonardder** [comment]
Agreed with @ehollig

**Comment by DrSooom** [comment]
@ehollig and @leonardder: After I have fully read issue #2257, I see no duplicate here. But just the issue description of issue #5197 is partly similar to this one here, but also not fully equal.

As NVDA knows all the time, which application is in the foreground, it shouldn't be such difficult to implement a routine to check this after an update was found. And only when "explorer.exe" is in the foreground the auto-update check notification will appear. Well, this new routine has to be forced every minute (or every five minutes), until the auto-update check notification was represented successfully. Hmm, that would need a new temporary stored value during a NVDA session. Or it could be written directly in the file "updateCheckState.pickle".

**Comment by Adriani90** [comment]
@DrSooom in my view if the update notification will be changed in the way that it will show up through Windows notifications in the system tray, then we do not need the apprearance at all and it will become obsolete anyway. At least in my view a notification in system tray which appears every few minutes or so is much simpler to implement. If you don't agree we can reopen this issue ofcourse.

**Comment by DrSooom** [comment]
@Adriani90: Maybe it would be better that I would have mentioned the obsolete state of this issue here, after #2257 has fixed, in my previous command as well. Of course you three are right, but as long as #2257 hasn't fixed, I personally wouldn't close this issue here.

Well, you also can label this issue here as "wontfix", because of the mentioned solutions in issue #2257. Then it would be fine to me – and comprehensible to all other users too. But "duplicate" would be still wrong in my opinion.
